### PR TITLE
Cap number of times error / conflict handler is invoked

### PR DESCRIPTION
**Pull request by shrishrirang**
- Closes https://github.com/Azure/azure-mobile-apps-js-client/issues/145
- Incorrectly setting pushError.isHandled = true can cause the error / conflict handling logic to go in an infinite loop
- This commit caps the number of times onError()/onConflict() is invoked for the same record.

### CODE REVIEW

**Comment by shrishrirang** [comment]
@danderson00

File: sdk/test/tests/target/cordova/push.tests.js [hunk 1]
```diff
@@ -57,7 +57,7 @@ $testGroup('push tests')
             id: '1',
             price: 2
         }).then(function() {
-            return pushAndValidateFeatures(false /* incremental sync */);
+            return pushAndValidateFeatures();
```
Comment (shrishrirang) [inline sdk/test/tests/target/cordova/push.tests.js]: Removing unnecessary params due to copy paste :)

File: sdk/src/sync/push.js [hunk 2]
```diff
@@ -50,6 +53,8 @@ function createPushManager(client, store, storeTaskRunner, operationTableManager
     // Resets the state for starting a new push operation
     function reset() {
         lastProcessedOperationId = -1; // Initialize to an invalid operation id
+        lastFailedOperationId = -1;
+        retryCount = 0;
```
Comment (danderson00) [inline sdk/src/sync/push.js]: Would it be better to create a new push manager for each push? There is a fair risk of introducing subtle bugs here by not cleaning up what needs to be cleaned up.

Comment (shrishrirang) [inline sdk/src/sync/push.js]: I can see the benefit of doing that. But we also want to be able to restrict number of concurrent push operations to 1. Using the same pushManager for all push operations helps keep the restriction an internal detail of the pushManager. If we start having multiple instances, callers of pushManager will have to co-ordinate to ensure only 1 instance is used at a time. That is something they need not have to worry about.

Comment (danderson00) [inline sdk/src/sync/push.js]: There are other, better ways of ensuring there is only one concurrent push. 

Single Responsibility Principle!

Comment (shrishrirang) [inline sdk/src/sync/push.js]: Ok, I'll talk to you about how to do this better. I plan to get this fix out tomorrow without making any major changes, so I'll take up the design changes separately.

**Pull request merged**